### PR TITLE
routing priority

### PR DESCRIPTION
**Pull request by ianlin-bbpos**
## Summary
Add routing priority picker.
Add routing priority field in createPaymentIntent method.

## Motivation
 To set the requested priority field when creating a PaymentIntent.

## Testing

- [x] I tested this manually
- [ ] I added automated tests

## Documentation

Select one:

- [ ] I have added relevant documentation for my changes.
- [x] This PR does not result in any developer-facing changes.

image

### CODE REVIEW

**Review by nazli-stripe — APPROVED** [review]

File: dev-app/src/screens/CollectCardPaymentScreen.tsx [hunk 1]
```diff
@@ -492,6 +504,29 @@ export default function CollectCardPaymentScreen() {
         </Picker>
       </List>
 
+      {discoveryMethod === 'internet' && (
```
Comment (nazli-stripe) [inline dev-app/src/screens/CollectCardPaymentScreen.tsx:507]: routing priority should be visible for all discovery methods, not just internet

**Review by nazli-stripe — CHANGES_REQUESTED** [review]

File: dev-app/src/screens/CollectCardPaymentScreen.tsx (hunk 1 above)
Comment (ianlin-bbpos) [reply dev-app/src/screens/CollectCardPaymentScreen.tsx:507]: got it~.

File: dev-app/src/screens/CollectCardPaymentScreen.tsx [hunk 2]
```diff
@@ -121,12 +129,16 @@ export default function CollectCardPaymentScreen() {
...
+    const routingPriority = {
+      requested_priority: inputValues.requestPriority,
+    };
     const paymentMethodOptions = {
       card_present: {
         request_extended_authorization:
           inputValues.requestExtendedAuthorization,
         request_incremental_authorization_support:
           inputValues.requestIncrementalAuthorizationSupport,
+        routing: routingPriority,
```
Comment (nazli-stripe) [inline dev-app/src/screens/CollectCardPaymentScreen.tsx:141]: this is currently not passed to SDK's createPaymentIntent method call for non Internet readers. Should update L193 on this file to include this option

**Review by nazli-stripe — CHANGES_REQUESTED** [review]
For non internet readers the SDK support creating a PaymentIntent on the client side (without calling Stripe backend). Native SDKs have support for passing the routing option but that seems to be lacking for RN SDK.

We should add mappers for iOS and Android for that parameter and update the types to include routing:

iOS: https://github.com/stripe/stripe-terminal-react-native/blob/8454a922/ios/StripeTerminalReactNative.swift#L376-L377

Android: https://github.com/stripe/stripe-terminal-react-native/blob/8454a922/ios/StripeTerminalReactNative.swift#L376-L377

PaymentMethodOptions type: https://github.com/stripe/stripe-terminal-react-native/blob/8454a922/src/types/index.ts#L171-L174

**Comment by nazli-stripe** [comment]
@ianlin-bbpos for reference here is the createPaymentIntent documentation for iOS and Android. 

Android: https://stripe.dev/stripe-terminal-android/core/com.stripe.stripeterminal/-terminal/create-payment-intent.html

iOS: https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPPaymentIntentParameters.html

**Review by nazli-stripe — APPROVED** [review]

**Pull request merged**